src/D2L.CodeStyle.Analyzers/UnsafeStaticsAnalyzer.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using D2L.CodeStyle.Analysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers {
+    [DiagnosticAnalyzer( LanguageNames.CSharp )]
+    public sealed class UnsafeStaticsAnalyzer : DiagnosticAnalyzer {
+
+        public const string DiagnosticId = "D2L0002";
+        private const string Category = "Safety";
+
+        private const string Title = "Ensure that static field is safe in undifferentiated servers.";
+        private const string Description = "Static fields should not have client-specific or mutable data, otherwise they will not be safe in undifferentiated servers.";
+        internal const string MessageFormat = "This static is unsafe because: '{0}'.";
+
+        private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId, Title, MessageFormat, Category,
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true,
+            description: Description

```

Comment:
Gonna nit: I hate mixing formatting.  Either put it all on different lines, or together